x-pack/plugins/uptime/common/types/index.ts

```diff
@@ -28,3 +30,17 @@ export interface MonitorDurationResult {
 export interface MonitorIdParam {
   monitorId: string;
 }
+
+export type SyntheticsMonitorSavedObject = SimpleSavedObject<{
```

Comment:
We should add all properties we currently support in the Integration. You can see which we support by reviewing the `manifest.yml` field in each data stream (icmp, TCP, HTTP, and browser - ignore browser.screenshoot and browser.network).

Reply:
I think this is just for type safety, instead of adding all the fields in one type I think we should create either conditional types or create a separate type for each monitor type. Since mix of optional fields can be confusing. I left it deliberately to be filled by UI.

This has no binding on what we can save in saved object though. That we will control via route validation